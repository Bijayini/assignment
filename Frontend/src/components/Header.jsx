import React, {Component} from 'react';
import { Link } from "react-router-dom";

import Button from "../ui-components/Button";

import CreateTask from "../containers/CreateTask";

class Header extends Component{
  constructor(){
    super();
    this.state = {
      isOverlayOpen:false
    }
  }

  toggleOverlay=()=>{
    this.setState({
      isOverlayOpen:!this.state.isOverlayOpen
    })
  };

  render(){
    const {isOverlayOpen} = this.state;
    return(
      <header className="header">
        <div className="header-content wrap">
          <Link className="brand" to="/">Todo List</Link>
          <Button variant="secondary" size="small" onClick={this.toggleOverlay}>Create Task</Button>
          {isOverlayOpen &&
          <CreateTask toggleOverlay={this.toggleOverlay}/>
          }
        </div>
      </header>);
  }
}

export default Header;