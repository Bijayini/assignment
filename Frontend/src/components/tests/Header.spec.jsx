import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('Header', () => {
  let component;

  beforeEach(()=>{
    component = shallow(<Header />);
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });

  it('should open modal on clicking on plus button', () => {
    component.find('Button').simulate('click');

    expect(component.instance().state.isOverlayOpen).toBe(true);
  });

});
