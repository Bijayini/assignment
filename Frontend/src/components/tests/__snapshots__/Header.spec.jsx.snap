// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match with snapshot 1`] = `
"<header className=\\"header\\">
  <h1>
    ToDo List
  </h1>
  <div tabIndex={0} onClick={[Function]} className=\\"add-task\\">
    +
  </div>
</header>"
`;
